File: solutions/cpp/two-fer/1/two_fer.cpp
#include "two_fer.h"

const std::string first = "One for ";
const std::string second = ", one for me.";

namespace two_fer
{
    std::string two_fer()
    {
        return first + "you" + second;
    }

    std::string two_fer(const std::string& name)
    {
        return first + name + second;
    }

} // namespace two_fer

File: solutions/go/two-fer/1/two_fer.go
// Package twofer implements a simple library for sharing things.
package twofer

// ShareWith returns a string with the format "One for [name], one for me."
// If the name is empty, it returns "One for you, one for me."
func ShareWith(name string) string {
	if name == "" {
        name = "you"
    }
	return "One for " + name + ", one for me."
}
